src/environment/environment.py

```diff
@@ -0,0 +1,165 @@
+import pygame
+import numpy as np
+from pygame.surface import Surface
+from typing import Tuple
+
+
+class Environment:
+    def __init__(
+        self, screen: Surface, agent_size: int = 10, render_on: bool = False
+    ) -> None:
+        """
+        Initializes the environment.
+
+        Args:
+            screen (Surface): Pygame screen surface.
+            agent_size (int, optional): Size of the agent. Defaults to 10.
+            render_on (bool, optional): Whether to render the environment. Defaults to False.
+        """
+        self.screen = screen
+        self.render_on = render_on
+        self.agent_size = agent_size
+
+        self.agent_location = (
+            (self.screen.get_size()[0] / 2),
+            (self.screen.get_size()[1] / 2),
+        )
+
+        self.rewards = {
+            "alive": 1,
+            "death": -100,
+            "not possible": -5,
+        }
+
+        self.actions = {
+            "up": 0,
+            "down": 1,
+            "left": 2,
+            "right": 3,
+        }
+
```

Comment:
Should we also run `self.reset()` upon init? Because now you need to run the reset function in order for the environment to be set, which feels a bit counter intuitive for me (as reset sounds like there already was a state that is now changed).